temp-name/src/main/java/org/jboss/bacon/tempname/impl/generator/BuildConfigGenerator.java

```diff
@@ -2,17 +2,43 @@
 
 import org.jboss.bacon.tempname.impl.config.BuildConfigGeneratorConfig;
 import org.jboss.bacon.tempname.impl.dependencies.DependencyResult;
+import org.jboss.bacon.tempname.impl.dependencies.Project;
+import org.jboss.bacon.tempname.impl.projectfinder.FoundProject;
 import org.jboss.bacon.tempname.impl.projectfinder.FoundProjects;
+import org.jboss.da.model.rest.GAV;
 import org.jboss.pnc.bacon.pig.impl.config.BuildConfig;
+import org.jboss.pnc.bacon.pig.impl.mapping.BuildConfigMapping;
 
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
 import java.util.List;
+import java.util.Set;
 
 public class BuildConfigGenerator {
     public BuildConfigGenerator(BuildConfigGeneratorConfig buildConfigGeneratorConfig) {
         throw new UnsupportedOperationException();
     }
 
     public List<BuildConfig> generateConfigs(DependencyResult dependencies, FoundProjects foundProjects) {
-        throw new UnsupportedOperationException();
+        List<BuildConfig> generatedConfigs = new ArrayList<>();
+
+        Set<Project> projects = dependencies.getTopLevelProjects();
+        Set<GAV> gavsToSearch = new HashSet<>();
+        Iterator<Project> setIterator = projects.iterator();
+        while (!projects.isEmpty() && setIterator.hasNext()) {
+            gavsToSearch.addAll(setIterator.next().getGavs());
+        }
+
+        Set<FoundProject> foundProjectSet = foundProjects.getFoundProjects();
+        Iterator<FoundProject> foundProjectIterator = foundProjectSet.iterator();
+        BuildConfig tempBC;
+        while (!foundProjectSet.isEmpty() && foundProjectIterator.hasNext()) {
+            tempBC = BuildConfigMapping.toBuildConfig(
```

Comment:
Also I think you would want to call the method with takes the BC revison too, so that we have the revison that is closest to the version we want to build.